### ISSUE TITLE
Critical error when using pelican_flickrtag with latest pelican version

### ISSUE THREAD

**Issue by swissel** [streeter/pelican-flickrtag#4]
I updated my version of pelican last week to 3.5.0. Now whenever I compile my website with the pelican_flickrtag enabled, I get the error:

CRITICAL: coercing to Unicode: need string or buffer, NoneType found
make: **\* [html] Error 1

I tried rolling back my pelican version, but still received the error. Pelican versions lower than 3.3.0 give other errors, but 3.4.0 and 3.3.0 also give the CRITICAL error.

What should I do?

**Comment by swissel** [comment]
I also tried running pelican with debug options:

bash-3.2$ pelican /Users/wissels/website/content -o /Users/wissels/website/output -s /Users/wissels/website/pelicanconf.py --debug -vvv
-> [flickrtag]: Parsing articles for photo ids...
-> [flickrtag]: Found 35 photo ids in the articles
CRITICAL: coercing to Unicode: need string or buffer, NoneType found

**Comment by icaoberg** [comment]
@swissel I just had the same issue and I think there is quick solution 

This issue is that according to the documentation 

```
FLICKR_TAG_CACHE_LOCATION - The cache location which stores the looked up 
photo information. This dramatically speeds up building of the site and permits you 
to do it offline as well. Defaults to /tmp/com.chrisstreeter.flickrtag-images.cache 
(Optional)
```

Unfortunately it is working on the assumption that the user has writing access to `/tmp`.

You can add this line to `pelicanconf.py`

```
FLICKR_TAG_CACHE_LOCATION=/<path_to_a_directory_you_can_write_to>/tempfile
```

I added the line 
```
FLICKR_TAG_CACHE_LOCATION="temp/flickrfiles"
```
and made sure the folder ```./temp``` existed.  It works now.

This is, more than likely, a small oversight. I hope this information helps. 

FYI if you run make with ```DEBUG=1``` it will give you more information about the process of making the static files

```
make DEBUG=1 html
```

**Comment by streeter** [comment]
Ah, good catch. Yeah, seems like the cache location isn't checked to be writeable. I'd assumed that `/tmp` would be writeable by users. For my project I use `FLICKR_TAG_CACHE_LOCATION = './com.chrisstreeter.flickrtag-images.cache'`.

**Comment by rossburton** [comment]
I just hit that, and I can write to /tmp.  The problem for me is that the default isn't being respected, if I put a print(tmp_file) before the file is opened then it prints None.  Assigning a different value in the config file simply ensures that it has a value, and this can be demonstrated by setting the default value in the config file.

**Comment by gpoo** [comment]
I can reproduce the `TypeError: coercing to Unicode: need string or buffer, NoneType found` issue by switching between Python 2.7 and Python 3.5, and having some non-ASCII character stored.

The patch in #6 should address that part of the issue.

**Comment by gpoo** [comment]
@rossburton That happens because `generator.context.get(...)` returns nothing. Using `generator.settings[...]` fixes the issue. See patch in #6.

**Comment by gpoo** [comment]
I think this can be safely closed, as #6 was already pushed.
